react-components/src/apps/quickLaunch/CreateQuickLaunchDialog.js

```diff
@@ -0,0 +1,107 @@
+/**
+ *
+ * @author sriram
+ *
+ */
+import React from "react";
+import withI18N, { formatMessage, getMessage } from "../../util/I18NWrapper";
+import { injectIntl } from "react-intl";
+import intlData from "../messages";
+import Dialog from "@material-ui/core/Dialog";
+import DEDialogHeader from "../../util/dialog/DEDialogHeader";
+import DialogContent from "@material-ui/core/DialogContent";
+import { ErrorMessage, Field, Form, Formik } from "formik";
+import {
+    FormCheckbox,
+    FormMultilineTextField,
+    FormTextField,
+} from "../../util/FormField";
+import Button from "@material-ui/core/Button";
+import DialogActions from "@material-ui/core/DialogActions";
+import { Tooltip } from "@material-ui/core";
+
+function CreateQuickLaunchDialog(props) {
+    const { dialogOpen, appName, presenter, intl } = props;
+
+    const handleSubmit = (values, actions) => {
+        console.log("submit clicked->" + values.name);
+        actions.setSubmitting(true);
+        const { name, description, is_public } = values;
+        presenter.createQuickLaunch(
+            name,
+            description ? description : "",
+            is_public ? is_public : false,
```

Comment:
I'm not familiar with Formik yet, but are the `description` and `is_public` checks necessary?  For example, if `is_public` is tied to a CheckBox, I would think it would automatically be either true or false.

Reply:
As far as my understanding, if a field is not filled by user, that field will not show up under `value` object. So we have to manually set those values to "" or false depending on the type of field. For example, if the user didn't check `Public` checkbox, then `values.is_public` will be `undefined` but i can't send that to java so I have to manually set it to `false`. May be @psarando has a different idea 🤔

Reply:
Ok after reading formik docs, it looks like, if i set initial values for `description` and `is_public`, then i don't have to set them here.